### PR TITLE
Update TwitterPreview to make hashtags and URLs clickable

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Related to https://github.com/Automattic/jetpack/pull/30029

## Proposed Changes

* Update TwitterPreview to make hashtags and URLs clickable

## Testing Instructions

* Follow the instructions given in https://github.com/Automattic/jetpack/pull/30029
* Add a custom message with hashtags and/or URLs
* Confirm that the hashtags and URLs are converted to hyperlinks

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204281809108915

### CODE REVIEW

**Comment by monsieur-z** [comment]
The code looks good!

Is it supposed to work when previewing posts in Calypso (at `/posts/:site` > _Share_ > _Preview_ > _Twitter_)?

**Comment by manzoorwanijk** [comment]
> Is it supposed to work when previewing posts in Calypso (at `/posts/:site` > _Share_ > _Preview_ > _Twitter_)?

I am not sure about that.

**Comment by gmjuhasz** [comment]
Kevin asked this on Slack and Kristina said it's not in scope, unless it's a quick change: p1681420114153609-slack-C04UC7YGPNY

**Comment by manzoorwanijk** [comment]
We can make that change in a separate PR.

File: packages/social-previews/test/index.js
Comment (gmjuhasz) [inline packages/social-previews/test/index.js:1]: Do you think we could add some tests for the hastags same way we have them for the URLs?

Comment (manzoorwanijk) [reply packages/social-previews/test/index.js:1]: Done. Thanks

**Review by gmjuhasz — APPROVED** [review]
Thanks for adding the extra tests! Looks good to me 👍

**Pull request merged**